Remove & upgrade dependencies from base image

francoisferrand opened the pull request:
- switch to the 'slim' variant, to avoid importing useless packages
- use multi-stage build to avoid including build tool in production image
- use `yarn install` to install packages

NOTE: this does not fix all issues, we would need to switch to different distribution for this (see #225 or #224), which would need better assessment of impact.

Issue: S3UTILS-69

bert-e commented:
# Hello francoisferrand,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/s3utils/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue S3UTILS-69 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `1.12.6`

* `1.13.0`


Please check the `Fix Version/s` of S3UTILS-69, or the target
branch of this pull request.

alexanderchan-scality commented:
@francoisferrand it seems like it would also be beneficial to use `yarn install` and perform clean up after.
something like:
```
RUN apt-get autoremove --purge -y python git build-essential \
    && rm -rf /var/lib/apt/lists/* 
```

francoisferrand committed:
Use slim debian image

Avoid using too many packages in base image.

Issue: S3UTILS-69

francoisferrand committed:
Remove build tools from prod image

Use multi-stage build to reduce the packages present in the production
image.

Issue: S3UTILS-69

francoisferrand committed:
Use yarn to install package

Issue: S3UTILS-69

francoisferrand committed:
Use GHA cache for docker

This improves caching and speed ups the build, while also simplifying
the workflow definition.

Issue: S3UTILS-69

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/S3UTILS-69` into
`development/1.12`
* [w/1.13/improvement/S3UTILS-69](https://github.com/scality/s3utils/compare/development/1.13...w/1.13/improvement/S3UTILS-69) will be merged into `development/1.13` (pull request #226)





*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

francoisferrand committed:
Misc workflow improvements

- Avoid duplicated builds by ignoring development and q branches
- Cancel earlier build of the same branch
- Create GitHub release after docker build
- Use GHA cache in release's docker build
- Cleanup name of workflows & jobs

Issue: S3UTILS-69

bert-e commented:
# History mismatch



Merge commit #807ddfc983d5674246604a6f229f5aa00e36cda3 on the integration branch
`w/1.13/improvement/S3UTILS-69` is merging a branch which is neither the current
branch `improvement/S3UTILS-69` nor the development branch
`development/1.13`.

It is likely due to a rebase of the branch `improvement/S3UTILS-69` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

francoisferrand commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/S3UTILS-69` into
`development/1.12`
* [w/1.13/improvement/S3UTILS-69](https://github.com/scality/s3utils/compare/development/1.13...w/1.13/improvement/S3UTILS-69) will be merged into `development/1.13` (pull request #227)





*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

francoisferrand commented:
/approve

bert-e commented:
# Unknown command



I didn't understand this comment by @francoisferrand:

> /appove

I don't know what `appove` means.

Please **edit** or **delete** the corresponding comment so I can move on.

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/1.12`

* :heavy_check_mark: `development/1.13`




There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/1.12`

* :heavy_check_mark: `development/1.13`




Please check the status of the associated issue S3UTILS-69.

Goodbye francoisferrand.

francoisferrand commented:
also fixes S3UTILS-72
